From be0433dc5e973d3341e8514d77557176a9c2cb9f Mon Sep 17 00:00:00 2001
From: Christopher Champagne <cjchampagne@habana.ai>
Date: Tue, 29 Apr 2025 21:49:46 +0300
Subject: [PATCH] [SW-222861] g3: pldm: mask pdrs by platform

Some PDRs should be supported on HL325 and not HL338.
In preboot, only Gaudi3 current power consumption, 54V domain PDR for
sensor ID 375 needs to be masked as of now.

Change-Id: Ie98013d1446cd16a71892b6af3add2ca2a474cee
Signed-off-by: Christopher Champagne <cjchampagne@habana.ai>
---

diff --git a/arch/gaudi3/common/ras/pldm/pldm_pdr.c b/arch/gaudi3/common/ras/pldm/pldm_pdr.c
index a8f50b6..cc3b671 100644
--- a/arch/gaudi3/common/ras/pldm/pldm_pdr.c
+++ b/arch/gaudi3/common/ras/pldm/pldm_pdr.c
@@ -1911,31 +1911,36 @@
 			},
 		},
 	}},
-	[PLDM_RIDX_SENSOR_CURRENT_SOC_54V_POWER] = { .compact_numeric_sensor = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_CURRENT_SOC_54V_POWER,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
-			.record_change_number = 0,
+	[PLDM_RIDX_SENSOR_CURRENT_SOC_54V_POWER] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
 		},
-		.terminus_handle = 0,
-		.sensor_id = PLDM_SID_CURRENT_SOC_54V_POWER,
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance_number = 1,
-		.container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
-		.unit_modifier = -3,
-		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
-		.range_field_support = {
-			.warning_high = 0,
-			.warning_low = 0,
-			.critical_high = 0,
-			.critical_low = 0,
-			.fatal_high = 0,
-			.fatal_low = 0,
-		},
-		.sensor_name = "currentSOC54VPower",
-	}},
+		.compact_numeric_sensor = {
+			.header = {
+				.record_handle = PLDM_RH_SENSOR_CURRENT_SOC_54V_POWER,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.sensor_id = PLDM_SID_CURRENT_SOC_54V_POWER,
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance_number = 1,
+			.container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
+			.unit_modifier = -3,
+			.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
+			.range_field_support = {
+				.warning_high = 0,
+				.warning_low = 0,
+				.critical_high = 0,
+				.critical_low = 0,
+				.fatal_high = 0,
+				.fatal_low = 0,
+			},
+			.sensor_name = "currentSOC54VPower",
+		}
+	},
 	[PLDM_RIDX_SENSOR_COMPOSITE_TEMPERATURE] = { .compact_numeric_sensor = {
 		.header = {
 			.record_handle = PLDM_RH_SENSOR_COMPOSITE_TEMPERATURE,
